### PR TITLE
Adding support for domain claiming in CLI

### LINKED ISSUE #2787: Users SBAT add a claimed URL in their metadata
### Expected Behavior

**Make the `DOMAIN` claim verifiable**:

As part of the creation of the claim, the CLI should indicate instructions to add the following TXT record to their DNS settings:

`celo-site-verification=${base64encode(signatureOfClaimWithAddress)}`

Add a function similar to the [keybase verify function](https://github.com/celo-org/celo-monorepo/blob/master/packages/contractkit/src/identity/claims/keybase.ts#L23), that will check for the signature on the TXT record by looking it up via the [`dns` package](https://nodejs.org/api/dns.html#dns_dns_resolvetxt_hostname_callback)

### Current Behavior

We have a `DOMAIN` claim that is not verifiable.

### PR DESCRIPTION
**Pull request by aaitor**
### Description

It includes the following CLI changes:

* Modify the account:claim-domain command printing the TXT record to add to the domain claimed
* Provide the new command identity:test-domain-claim verifying if a domain name has the right TXT record

### Tested

tested locally:

**Requesting the domain claim:**
```
aitor@tijuana:~/Projects/Celo/Upstream/celo-monorepo/packages/cli (aaitor/cli-domain-claim)$ bin/run account:claim-domain ~/Projects/conf/Celo/metadata-orco.json --domain orco.dev --from  0x088e43db20b5606aa5a87af922ebc785599cbb49
Read Metadata from /home/aitor/Projects/conf/Celo/metadata-orco.json... done
Add claim... done
Write Metadata to /home/aitor/Projects/conf/Celo/metadata-orco.json... done
Please add the following TXT record to your domain:
celo-site-verification=MHgxZDE3ZTgyZjM2ZWI4MjFmZTJjMGJkYWM4ODA4MTRiYWQ5ZmYzZTViN2IxODNjMDNhZDRiOTIyYzhkZGRhZTRmMGNjYWMxYTAxMjFlYTEzNjg0MmZmZmNiNmE4ZjA0ZjAxODFlNWFjNjI0ZGRkMzgxZTIwMzE4NzI4ZmEyZjFiOTFi
```
At this point is necessary to add the `celo-site-verification` TXT record to the domain.
 
**Verifying the domain claim:**
```
aitor@tijuana:~/Projects/Celo/Upstream/celo-monorepo/packages/cli (aaitor/cli-domain-claim)$ bin/run identity:test-domain-claim  ~/Projects/conf/Celo/metadata-orco.json --domain orco.dev --from  0x088e43db20b5606aa5a87af922ebc785599cbb49
Read Metadata from /home/aitor/Projects/conf/Celo/metadata-orco.json... done
Fetching domain orco.dev TXT records for verification
Reading TXT records for domain orco.dev
✓ TXT Record celo-site-verification correctly found in orco.dev
```

**Testing with an invalid domain**:
```
aitor@tijuana:~/Projects/Celo/Upstream/celo-monorepo/packages/cli (aaitor/cli-domain-claim)$ bin/run identity:test-domain-claim  ~/Projects/conf/Celo/metadata-orco.json --domain orco.com --from  0x088e43db20b5606aa5a87af922ebc785599cbb49   
Read Metadata from /home/aitor/Projects/conf/Celo/metadata-orco.json... done
Fetching domain orco.com TXT records for verification
Reading TXT records for domain orco.com
x Unable to find a valid celo-site-verification TXT record in orco.com domain
```

### Other changes

N/A

### Related issues

- Fixes #2787 

### Backwards compatibility

This change is backward compatible

### CODE REVIEW

File: packages/cli/src/commands/identity/test-domain-claim.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    }),
+  }
+  static args = ClaimCommand.args
+  static examples = [
+    'test-domain-claim ~/metadata.json --domain test.com --from 0x47e172F6CfB6c7D01C1574fa3E2Be7CC73269D95',
+  ]
+  self = TestDomainClaim
+  async run() {
+    const res = this.parse(TestDomainClaim)
+    const metadata = this.readMetadata()
```
Comment (diminator) [inline packages/cli/src/commands/identity/test-domain-claim.ts:25]: How does this work with `metadata.json` of an account you don't control but want to verify?

**Review by diminator — CHANGES_REQUESTED** [review]
has this been tested against a real TXT record containing
`'celo-site-verification=' + signatureBase64`

**Comment by mrsmkl** [comment]
Verifying domain claims should be added to contractkit https://github.com/celo-org/celo-monorepo/blob/master/packages/contractkit/src/identity/claims/verify.ts#L18
Then `account:get-metadata` should work for showing if everything is verified or not.

**Comment by aaitor** [comment]
> has this been tested against a real TXT record containing
> `'celo-site-verification=' + signatureBase64`

yes, was tested with a real domain

File: packages/cli/src/commands/identity/test-domain-claim.ts (hunk 1 above)
Comment (aaitor) [reply packages/cli/src/commands/identity/test-domain-claim.ts:25]: metadata is off-chain and the urls publicly available in blockscout, so you can get that and validate

**Comment by aaitor** [comment]
Closed in favor of #3156

**Pull request closed without merging**